zerver/tests/test_events.py

```diff
@@ -257,6 +257,15 @@ def setUp(self) -> None:
         super().setUp()
         self.user_profile = self.example_user("hamlet")
 
+    def refresh_user_profile(self) -> None:
+        # Call this to make sure you don't have a stale UserProfile
+        # object after doing some kind of setup on the user before
+        # some action that acts on self.user_profile.
+        #
+        # Most of our callers are probably calling this for legacy
+        # reasons that no longer apply, but it's cheap enough.
+        self.user_profile = self.example_user("hamlet")
```

Comment:
Hmm, does `self.example_user()` use caches, though? Based on the comments, I think we really do want to make sure we're getting the user fresh. 

https://github.com/zulip/zulip/pull/11861 may or may not be relevant background for this detail; I remember that effort with @pragatiagrawal31 being surprisingly tangly in this test.

Reply:
It definitely uses caches!  That's the point.

By "fresh" I mean not going through Django's reload  mechanism.  Naming debates are welcome here. I am not completely happy with "fresh" or "refresh", although I think they're fine enough.

Reply:
#11861 is basically a case study in why **not** to use `refresh_from_db`.  Its semantics are not only confusing, but they're also not representative of what the app actually does.  We use `refresh_from_db` almost never in the app, and for good reason.